### ISSUE TITLE
Execute arbitrary Node.js code from within HTML

### ISSUE THREAD

**Issue by lastmjs** [tmpvar/jsdom#1658]
I would like to allow Node.js code to be called from within the HTML source. This is in an effort to support universal web components, the goal being to allow web components to render and run on all platforms. I've got some things working [here](https://github.com/scramjs/express-web-components) and [here](https://github.com/scramjs/scram-engine) if you want to look. Anyway, I came across this in the README:

> That is, code running inside the DOM's <script>s can, if it tries hard enough, get access to the Node environment, and thus to your machine.

How would one go about doing that? Also, would it be possible to remove the sandbox as an option so that scripts within the HTML/DOM context can call out to the Node.js environment? This could be immensely useful, as it would not only allow a headless browser, but even something like a headless Electron.

**Comment by lastmjs** [comment]
I've been poking around, and I've figured out a hack to get what I want. It looks like code within scripts are executed in their own V8 context, as described [here](https://nodejs.org/api/vm.html#vm_executing_javascript). If I augment the window object which is acting as the sandbox argument in this file `sdom/lib/jsdom/living/nodes/HTMLScriptElement-impl.js` in the `processJavaScript` function, it seems like I can call whatever Node.js code that I want. I attached `require` to the `window` object and was able to `require('fs')` and read from a file.

Can we allow this kind of behavior to be turned on and off?
